types/durableClient.d.ts

```diff
@@ -280,3 +235,109 @@ export declare class HttpManagementPayload {
      */
     readonly purgeHistoryDeleteUri: string;
 }
+
+/**
+ * Options object passed to client `getStatus()` method
+ */
+export interface GetStatusOptions {
+    /**
+     * Specifies whether execution history should be included in the response.
+     */
+    showHistory?: boolean;
+    /**
+     * Specifies whether input and output should be included in the execution history response.
+     */
+    showHistoryOutput?: boolean;
+    /**
+     * Specifies whether orchestration input should be included in the response.
+     */
+    showInput?: boolean;
+}
+
+/**
+ * Options object passed to DurableClient.raiseEvent()
+ */
+export interface RaiseEventOptions extends TaskHubOptions {
+    /**
+     * The ID of the orchestration instance that will handle the event.
+     */
+    instanceId: string;
+    /**
+     * The name of the event.
+     */
+    eventName: string;
+    /**
+     * The JSON-serializable data associated with the event.
+     */
+    eventData: unknown;
+}
+
+export interface SignalEntityOptions extends TaskHubOptions {
+    /**
+     * The name of the operation.
+     */
+    operationName?: string;
+    /**
+     * The content for the operation
+     */
+    operationContent?: unknown;
+}
+
+/**
+ * Options object passed to DurableClient APIs to specify task hub properties
+ */
+export interface TaskHubOptions {
+    /**
+     * The TaskHubName of the orchestration
+     */
+    taskHubName?: string;
+    /**
+     * The name of the connection string associated with `taskHubName.`
+     */
+    connectionName?: string;
+}
+
+/**
+ * Options object passed to DurableClient APIs
+ * to select certain orchestrations on which to perform
+ * actions
+ */
+export interface SelectionOptions {
```

Comment:
This name isn't super clear to me. What if we rename it to `OrchestrationFilter`?

Reply:
I believe this may actually be able to return Entity data so I would rather keep this a bit more general.
Selection might be a bit too broad though. Does `QueryOptions` seem better? After all, this is affecting the instance-querying APIs

Reply:
I'm not a fan of calling this `Options` because it's a required argument to both `getStatusBy` and `purgeInstanceHistoryBy`. If you like `Query` better than `Filter`, I'd be okay with `OrchestrationQuery`

If we do an `Options` object for this method, I think it should be a second argument. Options arguments typically only have optional/static information that affects how the method works across all method calls, but nothing really unique to any particular method call. For example, you might you use the same `timeout` value for all calls, but the `createdTimeFrom` will change often based on each particular call.

> I believe this may actually be able to return Entity data so I would rather keep this a bit more general.

I was assuming it was orchestration-only because of `runtimeStatus?: OrchestrationRuntimeStatus[];`, plus all the comments mentioning `orchestrations`. We could name this just `Query` or `Filter`, but I guess that feels too generic to me

Reply:
I looked at the C# API and it appears this only returns orchestration data, so I take back my comment on Entities.

I'm good with `OrchestrationFilter`, so let's go for it.

Reply:
Done, renamed to OrchestrationFilter